src/common/path.cc

```diff
@@ -11,6 +22,35 @@
 
 namespace BoBRobotics {
 namespace Path {
+filesystem::path
+getProgramFolder()
+{
+    return getProgramPath().parent_path();
+}
+
+filesystem::path
+getProgramPath()
+{
+#ifdef __linux__
+    char path[PATH_MAX + 1];
+    ssize_t len = readlink("/proc/self/exe", path, PATH_MAX);
+    BOB_ASSERT(len >= 0);
+    path[len] = '\0';
+#endif
+#ifdef _WIN32
```

Comment:
maybe an 
```c++
#ifdef __linux__
    // linux
#elif defined(_WIN32)
    // windows
#elif defined(__APPLE__)
    // mac
#else
    #error "Unsupported platform"
#endif
```
might be good

Reply:
Hmm... Yeah it is a bit clunky how it is.